src/storage/redis_db.h

```diff
@@ -33,12 +33,19 @@ class Database {
  public:
   static constexpr uint64_t RANDOM_KEY_SCAN_LIMIT = 60;
 
+  struct GetOptions {
+    // If snapshot is not nullptr, read from the specified snapshot,
+    // otherwise read from the "latest" snapshot.
+    const rocksdb::Snapshot *snapshot = nullptr;
+  };
+
   explicit Database(engine::Storage *storage, std::string ns = "");
   [[nodiscard]] rocksdb::Status ParseMetadata(RedisTypes types, Slice *bytes, Metadata *metadata);
-  [[nodiscard]] rocksdb::Status GetMetadata(RedisTypes types, const Slice &ns_key, Metadata *metadata);
-  [[nodiscard]] rocksdb::Status GetMetadata(RedisTypes types, const Slice &ns_key, std::string *raw_value,
-                                            Metadata *metadata, Slice *rest);
-  [[nodiscard]] rocksdb::Status GetRawMetadata(const Slice &ns_key, std::string *bytes);
+  [[nodiscard]] rocksdb::Status GetMetadata(GetOptions options, RedisTypes types, const Slice &ns_key,
```

Comment:
Can we add a simple comment here?
GetMetadata is an API that most new developers will use. The longgggg list of parameters can be overwhelming for them. Adding a simple comment can help boost their confidence in writing code.

Reply:
Updated. I'm not so good at doc writing, so @PragmaTwice @git-hulk @torwig @jihuayu would you mind take a look or edit that?

Reply:
Thanks, I think it is good enough